lib/src/rules/avoid_function_literals_in_foreach_calls.dart

```diff
@@ -16,6 +16,9 @@ const _details = r'''
 
 **AVOID** using `forEach` with a function literal.
 
+`forEach` requires a lambda, which creates a new scope, which increases code size and run time. 
```

Comment:
At a minimum, a new call stack. A new scope *if* an anonymous function is used.

@munificent likely has the right wording here...

Reply:
er, new stack *frame*

Reply:
It tends to be a fool's errand to rely on performance arguments for style choices because who know what the optimizing compiler is able to eliminate? In practice, I don't even know if `forEach()` is actually slower.

There are some subtle semantic reasons to think about:

* `return` inside the closure only returns from that iteration and not the surrounding function. That might be what you want, or might be confusing.
* If you assign to other variables inside the closure, it might break their ability to be type promoted where using a for-in loop is more visible to flow analysis.
* You can't `await` inside the closure.

But, for the most part, the main reason for this guideline is just consistency. You could mostly use `forEach()` without any problems, but there's no reason to prefer that over for-in and if there are two mostly equal choices, the style guide tends to just pick one because consistency is always valuable.

Reply:
I've had a go at summarising @munificent's exposition. I'm unsure if I've done it justice, or just butchered it.

On the flipside, as pointed out by @puf, the **good** code sample actually contained code from the **bad** code, which was causing confusion